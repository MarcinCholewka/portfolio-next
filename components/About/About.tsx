"use client";

import React from "react";
import { motion } from "framer-motion";
import { useTranslations } from "next-intl";

import { SectionHeading } from "@/components/SectionHeading";

export const About = () => {
  const t = useTranslations("About");

  return (
    <motion.section
      className="max-w-[45rem] text-center leading-8"
      initial={{ opacity: 0, y: 100 }}
      animate={{ opacity: 1, y: 0 }}
      transition={{ delay: 0.175 }}
    >
      <SectionHeading>{t("heading")}</SectionHeading>
      <p className="mb-3 ">{t("mainInfo")}</p>
      <p>{t("hobby")}</p>
    </motion.section>
  );
};
