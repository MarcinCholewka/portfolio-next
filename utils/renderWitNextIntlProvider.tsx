import { NextIntlProvider } from 'next-intl';
import { ReactElement } from 'react';
import { render as rtlRender, RenderOptions, RenderResult } from '@testing-library/react';

export const renderWitNextIntlProvider = (
  ui: ReactElement,
  {
    locale = 'en',
    messages = {},
    ...renderOptions
  }: RenderOptions & { locale?: string; messages?: {} } = {}
): RenderResult => {
  function Wrapper({ children }: { children?: React.ReactNode }) {
    return (
      <NextIntlProvider locale={locale} messages={messages}>
        {children}
      </NextIntlProvider>
    );
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
};
