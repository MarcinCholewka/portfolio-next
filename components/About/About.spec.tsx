import React from 'react';

import { About } from '@/components/About';
import { renderWitNextIntlProvider } from '@/utils/renderWitNextIntlProvider';
import messagesEN from '@/messages/en.json';

import '@testing-library/jest-dom';

describe('<About />', () => {
  it('renders without crashing', () => {
    const { container, getByText } = renderWitNextIntlProvider(<About />, {
      locale: 'en',
      messages: {
        About: messagesEN.About,
      },
    });

    expect(container).toMatchSnapshot();
    expect(getByText(messagesEN.About.heading)).toBeInTheDocument();
    expect(getByText(messagesEN.About.hobby)).toBeInTheDocument();
    expect(getByText(messagesEN.About.mainInfo)).toBeInTheDocument();
  });
});
