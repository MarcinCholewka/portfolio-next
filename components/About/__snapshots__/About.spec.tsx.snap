// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> renders without crashing 1`] = `
<div>
  <section
    class="max-w-[45rem] text-center leading-8"
    style="opacity: 0; transform: translateY(100px) translateZ(0);"
  >
    <h2
      class="text-3xl text-center font-medium capitalize mb-8"
    >
      About me
    </h2>
    <p
      class="mb-3 "
    >
      I am a front-end developer with in-depth knowledge of React.js and its core principles, as well as front-end layout styling. I have strong communication skills, characterized by excellent interpersonal effectiveness in individual and team interactions. Over the last few years, I have gained experience in designing and implementing high-quality, elegant SaaS software for web and mobile applications. My core stack is React, but I also know TypeScript. I always want to learn new technologies. I am currently looking for a full-time job as a front-end developer.
    </p>
    <p>
      When I'm not coding, I spend time with my family. I like riding motorcycles and I'm constantly training my riding skills. I started practicing karate with my son, which makes me feel better when doing everyday activities. I also like learning new things. I'm currently trying to expand my knowledge of Node.js.
    </p>
  </section>
</div>
`;
